python/ray/train/v2/_internal/execution/controller/controller.py

```diff
@@ -476,11 +476,16 @@ async def run(self):
         # TODO: move to __del__ after https://github.com/ray-project/ray/issues/53169
         self._shutdown()
 
+        # Call after_controller_finish with the final result
+        result = self._build_result()
+        for callback in self._controller_callbacks:
+            callback.after_controller_finish(result)
+
     async def abort(self):
         """Trigger callback abort hooks and terminate the controller process."""
-        self._set_state(AbortedState())
         if self._worker_group:
             self._worker_group.abort()
+        self._set_state(AbortedState())
```

Comment:
can you add a comment about the ordering here?

Reply:
Done.